src/recorder/jamrecorder.cpp

```diff
@@ -77,7 +78,7 @@ void CJamClient::Frame(const QString _name, const CVector<int16_t>& pcm, int iSe
 
     for(int i = 0; i < numChannels * iServerFrameSizeSamples; i++)
     {
-        *out << pcm[i];
+        if (out) *out << pcm[i];
```

Comment:
This shouldn't be necessary - it looks like it's losing data.  That mustn't happen.

Reply:
I just don't like to follow a pointer unless I am certain it is non-null. If the constructor throws an exception, `out` might not be set. Just belt and braces really - defensive programming.

Reply:
That needs to be ensured elsewhere.  This mustn't be called without `*out` being valid - if it's been freed elsewhere, someone broke something.

Reply:
It shouldn't be freed anywhere else except in `CJamClient::Disconnect()`. And maybe the check here isn't necessary if it would never be called unless `CJamClient::CJamClient()` succeeds without an exception. Happy to remove the test, as the code has been there a long time and there have been no reports of any crashes due to using a null `out` here. Perhaps if we do want to keep the test it should be in the three `operator <<` methods in `cwavestream.cpp`. But even that may be overkill, if none of these functions will ever be called unless `CJamClient()` succeeded.

Reply:
Yup - it's my "It should just work, hasn't appeared broken, don't change it" approach :).